econml/utilities.py

```diff
@@ -697,3 +751,124 @@ def predict(self, Xt):
+        Whether to fit an intercept
+
+    Attributes
+    ----------
+    fit_args: dict of str: object
+        The arguments to pass to the `OLS` regression's `fit` method.  See the
+        statsmodels documentation for more information.
+    results: RegressionResults
+        After `fit` has been called, this attribute will store the regression results.
+    """
+
+    def __init__(self, fit_intercept=False):
+        self.fit_args = {}
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit the ordinary least squares model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+        y : array_like, shape (n_samples, 1) or (n_samples,)
+            Target values
+        sample_weight : array_like, shape (n_samples,)
+            Individual weights for each sample
+
+        Returns
+        -------
+        self
+        """
+        assert ndim(y) == 1 or (ndim(y) == 2 and shape(y)[1] == 1)
+        y = reshape(y, (-1,))
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        if sample_weight is not None:
+            ols = WLS(y, X, weights=sample_weight, hasconst=self.fit_intercept)
+        else:
+            ols = WLS(y, X, hasconst=self.fit_intercept)
+        self.results = ols.fit(**self.fit_args)
+        return self
+
+    def predict(self, X):
+        """
+        Predict using the model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Samples.
+
+        Returns
+        -------
+        array, shape (n_samples,)
+            Predicted values
+        """
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        return self.results.predict(X)
+
+    def predict_interval(self, X, alpha):
+        """
+        Get a confidence interval for the prediction at `X`.
+
+        Parameters
+        ----------
+        X : array-like
+            The features at which to predict
+        alpha : float
+            The significance level to use for the interval
+
+        Returns
+        -------
+        array, shape (2, n_samples)
+            Lower and upper bounds for the confidence interval at each sample point
+        """
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        # NOTE: we use `obs = False` to get a confidence, rather than prediction, interval
+        preds = self.results.get_prediction(X).conf_int(alpha=alpha, obs=False)
+        # statsmodels uses the last dimension instead of the first to store the confidence intervals,
+        # so we need to transpose the result
+        return transpose(preds)
+
+    @property
+    def coef_(self):
+        return self.results.params
+
+    def coef__interval(self, alpha):
+        return transpose(self.results.conf_int(alpha=alpha))
+
+
+class LassoCVWrapper:
```

Comment:
I think to avoid the warning about y.ravel() when y is 2-d by y.shape[1]=1, this wrapper, should be doing the ravel when y is 2-d and y.shape[1]=1 and then unravel at predict time.

Reply:
This will be addressed in the next push.